corehq/apps/app_manager/static/app_manager/js/app_manager.js

```diff
@@ -141,6 +150,34 @@ hqDefine('app_manager/js/app_manager.js', function () {
         module.setCommcareVersion(args.commcareVersion);
     };
 
+    var _initPublishStatus = function () {
+        var currentAppVersionUrl = hqImport('hqwebapp/js/initial_page_data.js').get('current_app_version_url');
+        var _checkPublishStatus = function () {
+            $.ajax({
+                url: currentAppVersionUrl,
+                success: function (data) {
+                    module.setPublishStatus((!data.latestRelease && data.currentVersion > 1) || (data.latestRelease !== null && data.latestRelease < data.currentVersion));
+                }
+            });
+        };
+        _checkPublishStatus();
+        // check publish status every 20 seconds
+        setInterval(_checkPublishStatus, 20000);
```

Comment:
This could probably be less frequent given that the code is also checking for known app-modifying ajax calls below.